extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTransitSecretEngine.java

```diff
@@ -0,0 +1,92 @@
+package io.quarkus.vault;
+
+import java.util.List;
+import java.util.Map;
+
+import io.quarkus.vault.transit.ClearData;
+import io.quarkus.vault.transit.DecryptionRequest;
+import io.quarkus.vault.transit.EncryptionRequest;
+import io.quarkus.vault.transit.RewrappingRequest;
+import io.quarkus.vault.transit.SigningInput;
+import io.quarkus.vault.transit.SigningRequest;
+import io.quarkus.vault.transit.TransitContext;
+import io.quarkus.vault.transit.VerificationRequest;
+
+public interface VaultTransitSecretEngine {
+
+    // encrypt
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * equivalent to:
+     * {@code encrypt(keyName, ClearData.from(clearData), null);}
+     * <p>
+     * this method is usually used in conjunction with {@link #decrypt(String, String)}
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the string to encrypt
+     * @return cipher text
+     */
+    String encrypt(String keyName, String clearData);
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * if the key does not exist, and the policy specifies a create capability the key will be lazily created
+     * (i.e. upsert). the key can be further customized by specifying transit encryption-key configuration
+     * properties.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the data to encrypt
+     * @param transitContext an optional context for key derivation
+     * @return cipher text
+     */
+    String encrypt(String keyName, ClearData clearData, TransitContext transitContext);
+
+    /**
+     * Encrypt a list of elements. This will return a list of cipher text.
+     * Each element can specify an optional context, allowing key derivation.
+     * If one or more encryptions fail, a VaultEncryptionBatchException will be thrown with the full
+     * detail of all operations that succeeded, and all that failed, with associated error messages.
+     *
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param requests the list of elements to encrypt
+     * @return cipher texts
+     */
+    Map<EncryptionRequest, String> encrypt(String keyName, List<EncryptionRequest> requests);
+
+    // decrypt
```

Comment:
Please complete the javadocs for all the other methods and remove the comments like `// decrypt` etc as the method names are self-descriptive

Reply:
done